Hexastore.Rocks/RocksGraphProvider.cs

```diff
@@ -39,61 +52,93 @@ public RocksGraphProvider(ILogger<RocksGraphProvider> logger, string path = null
             .IncreaseParallelism(6)
             .SetCompression(Compression.No);
             */
-
-            _db = RocksDb.Open(options, dataPath);
-            _logger.LogInformation($"created rocksdb at {dataPath}");
         }
 
+
         public bool ContainsGraph(string id, GraphType type)
         {
             var key = MakeKey(id, type);
-            return _db.Get(key) == "1";
+            var db = GetOrOpenDB(id);
+            return db.Get(key) == "1";
         }
 
         public IStoreGraph CreateGraph(string id, GraphType type)
         {
             var key = MakeKey(id, type);
-            _db.Put(key, "1");
-            return new RocksGraph(key, _db);
+            var db = GetOrOpenDB(id);
+            db.Put(key, "1");
+            return new RocksGraph(key, db);
         }
 
         public bool DeleteGraph(string id, GraphType type)
```

Comment:
we probably need a call from a higher level to delete a 'store' that would be a cleanup on the filesystem. I have left that item pending for while.

Reply:
Good catch, there are some scenarios here that should be a check only. I'll fix this.